master/buildbot/steps/source/git.py

```diff
@@ -74,11 +74,16 @@ class Git(Source, GitStepMixin):
     def __init__(self, repourl=None, branch='HEAD', mode='incremental', method=None,
                  reference=None, submodules=False, shallow=False, progress=False, retryFetch=False,
                  clobberOnFailure=False, getDescription=False, config=None,
-                 origin=None, sshPrivateKey=None, sshHostKey=None, sshKnownHosts=None, **kwargs):
+                 origin=None, sshPrivateKey=None, sshHostKey=None, sshKnownHosts=None, supportsProgress=None, **kwargs):
```

Comment:
This param exposed to the user, but one shouldn't need to specify if the git on the worker supports the flag. Better, just pass `progress=True` by default and don't add the `--progress` flag if it turns out that we don't support progress. A warning should be logged in that case.